src/MoviePage/MoviePage.js

```diff
@@ -1,21 +1,63 @@
 import React, {Component} from "react";
-import {getIndividualMovie} from "../api";
+import {getIndividualMovie, postUserRating} from "../api";
 
 class MoviePage extends Component {
   constructor(props) {
-    super();
+    super(props);
+    console.log(this.props);
     this.state = {
       movie: {},
       error: "",
       movieId: props.params.id,
+      userRating: "Not yet rated",
+      displayedRating: 0,
+      userId: typeof props.userId === "number" ? props.userId : null,
     };
   }
   async componentDidMount() {
+    let userRating;
+    if (this.state.userId) {
+      userRating = this.props.userRatings.find(
+        (rating) => parseInt(rating.movie_id) === parseInt(this.state.movieId)
+      );
+      if (userRating) {
+        userRating = userRating.rating;
+      }
+    }
     const response = await getIndividualMovie(this.state.movieId);
-    console.log(response)
     const movie = await response.movie;
-    this.setState({movie});
+    this.setState({movie, userRating});
   }
+  rateMovie = (e) => {
+    const rating = parseInt(e.target.value);
+    this.setState({displayedRating: rating});
+  };
+  submitRating = async (e) => {
+    e.preventDefault();
+    let response = await postUserRating(
+      this.state.userId,
+      this.state.displayedRating,
+      this.state.movie.id
+    );
+    if (response.error) {
+      let userInput = window.confirm(
+        "You allready have a rating for this movie, would you like to delete it?"
+      );
+      if (userInput) {
+        this.deleteMovie();
+      } else {
+        this.setState({userRating: this.state.displayedRating});
+      }
+    } else {
+      this.setState({userRating: this.state.displayedRating});
+    }
+
+    console.log(response);
+  };
+  deleteMovie = () => {
```

Comment:
Think this was a good refactor - it will likely be better UX to not have so many buttons around one feature